libminifi/src/utils/RegexUtils.cpp

```diff
@@ -22,58 +22,86 @@
 #include <vector>
 
 #include "Exception.h"
+#include <regex.h>
+
+namespace org::apache::nifi::minifi::utils {
 
 #ifndef NO_MORE_REGFREEE
-namespace {
 
-std::size_t getMaxGroupCountOfRegex(const std::string& regex) {
-  return std::count(regex.begin(), regex.end(), '(') + 1;
+SMatch::SMatch(const SMatch& other) {
+  *this = other;
 }
 
-}  // namespace
-#endif
-
-namespace org::apache::nifi::minifi::utils {
+SMatch::SMatch(SMatch&& other) {
+  *this = std::move(other);
+}
 
-#ifndef NO_MORE_REGFREEE
-SMatch::SuffixWrapper SMatch::suffix() const {
-  if ((size_t) matches_[0].match.rm_eo >= string_.size()) {
-    return SuffixWrapper{std::string()};
-  } else {
-    return SuffixWrapper{string_.substr(matches_[0].match.rm_eo)};
+SMatch& SMatch::operator=(const SMatch& other) {
+  if (this == &other) {
+    return *this;
+  }
+  reset(other.string_);
+  matches_.reserve(other.matches_.size());
+  ready_ = other.ready_;
+  for (const auto& sub_match : other.matches_) {
+    size_t begin_off = gsl::narrow<size_t>(std::distance(other.string_.begin(), sub_match.first));
+    size_t end_off = gsl::narrow<size_t>(std::distance(other.string_.begin(), sub_match.second));
+    matches_.push_back(Regmatch{sub_match.matched, string_.begin() + begin_off, string_.begin() + end_off});
```

Comment:
`clang-tidy` wants this changed to `emplace_back`, and although it looks like this check is not turned on yet, it will be eventually, so let's change it (also in a few other places in this file)

Reply:
changed these to `emplace_back`